apps/src/studio/StudioAnimation.js

```diff
@@ -0,0 +1,202 @@
+     this.spriteSheet_.assetPath);
+  this.element_.setAttribute('id', itemId);
+  this.element_.setAttribute('height',
+      this.spriteSheet_.assetHeight() * this.renderScale_);
+  this.element_.setAttribute('width',
+      this.spriteSheet_.assetWidth() * this.renderScale_);
+  parentElement.appendChild(this.element_);
+
+  this.element_.setAttribute('clip-path', 'url(#' + clipId + ')');
+};
+
+/**
+ * Remove our element/clipPath/animator
+ */
+StudioAnimation.prototype.removeElement = function() {
+
+  if (this.element_) {
+    this.element_.parentNode.removeChild(this.element_);
+    this.element_ = null;
+  }
+
+  // remove clip path element
+  if (this.clipPath_) {
+    this.clipPath_.parentNode.removeChild(this.clipPath_);
+    this.clipPath_ = null;
+  }
+
+  if (this.animator_) {
+    window.clearInterval(this.animator_);
+    this.animator_ = null;
+  }
+};
+
+/**
+ * Display the current frame at the given location
+ */
+StudioAnimation.prototype.redrawCenteredAt = function (center) {
+  var frame = this.spriteSheet_.getFrame(this.currentAnimation_, this.currentFrame_);
+  var scale = this.renderScale_;
+
+  // Preserved behavior: When scaling a sprite up, we actually scale around the
+  //       bottom-center of the sprite (so feet stay planted in the same place)
+  //       rather than actually around its center.
+  //       That's what the (2 * scale - 1) bit is about; just change that to
+  //       (scale) if you want to scale about the sprite center again.
+  // TODO: Improve this by scaling around an explicitly encoded 'sprite center'
+  var topLeft = {
+    x: center.x - (frame.width / 2) * scale,
+    y: center.y - (frame.height / 2) * (2 * scale - 1)
+  };
+
+  // Offset the spritesheet DOM element by the inverse of the offset of the
+  // frame we want to display.
+  this.element_.setAttribute('x', topLeft.x - frame.left * scale);
+  this.element_.setAttribute('y', topLeft.y - frame.top * scale);
+  this.element_.setAttribute('opacity', this.opacity_);
+
+  // Then set the clip rect to the position where we want to display it, so
+  // only the frame that's now positioned correctly is shown.
+  var clipRect = this.clipPath_.childNodes[0];
+  clipRect.setAttribute('x', topLeft.x);
+  clipRect.setAttribute('y', topLeft.y);
+};
+
+/**
+ * Sets which animation to play out of the sprite sheet.
+ * Animations are indexed by their position in the sprite sheet, where each
+ * animation is its own column and animation zero is the far-left column.
+ * @param {!number} animationIndex
+ */
+StudioAnimation.prototype.setCurrentAnimation = function (animationIndex) {
+  this.currentAnimation_ = animationIndex;
+};
+
+/**
+ * Set the animation rate for this item's sprite.
+ * @param {number} framesPerSecond
+ */
+StudioAnimation.prototype.setAnimationRate = function (framesPerSecond) {
+  if (this.animator_) {
+    window.clearInterval(this.animator_);
+  }
+  this.animator_ = window.setInterval(function () {
```

Comment:
Interesting case here - we update `currentFrame_` on setInterval, but the sprite only redraws when `redrawCenteredAt` is called manually.  When these are out of sync - and they _are_ out of sync, because studio tries to tick at 30FPS but none of our animations run at multiples of that speed - you can get odd frame skips occasionally.

Not a big issue and I left it in to preserve current behavior.  Just something we should maybe do differently in the future.